### DISCUSSION TITLE
Collecting page data .SyntaxError: Unexpected token u in JSON at position 0

### DISCUSSION THREAD

**Discussion by coding-to-music** [vercel/next.js#43833] [Help]
Hello, my repo is here: https://github.com/coding-to-music/swejobs-next-firebase

I am attempting to build a project I found on GitHub

From / By https://github.com/kevinshen56714/SWEJobs.fyi

https://swejobs.fyi/ (pretty good site)

I am trying to do a build and I get an error trying to collect page data - so not sure if it is something in my setup or if the people who wrote the original repo ran into similar issues.

I am unable to run a build locally and also Vercel has the error too.

I posted the issue originally here: https://github.com/vercel/vercel/discussions/9006

## Error creating a build

```
npm run build
```

Output

```
> swejobs-next-firebase@0.1.0 build
> next build

info  - Loaded env from /mnt/volume_nyc1_01/swejobs-next-firebase/.env
info  - Linting and checking validity of types
info  - Creating an optimized production build
info  - Compiled successfully
info  - Collecting page data .SyntaxError: Unexpected token u in JSON at position 0
    at JSON.parse (<anonymous>)
    at /mnt/volume_nyc1_01/swejobs-next-firebase/.next/server/chunks/186.js:961:27

> Build error occurred
Error: Failed to collect page data for /about/about-data
    at /mnt/volume_nyc1_01/swejobs-next-firebase/node_modules/next/dist/build/utils.js:959:15
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  type: 'Error'
}
info  - Collecting page data .
```

## Error when Vercel tries to do a build

```
info  - Linting and checking validity of types...
info  - Creating an optimized production build...
info  - Compiled successfully
info  - Collecting page data...
SyntaxError: Unexpected token u in JSON at position 0
    at JSON.parse (<anonymous>)
    at /vercel/path0/.next/server/chunks/186.js:961:27
> Build error occurred
Error: Failed to collect page data for /about/about-data
    at /vercel/path0/node_modules/next/dist/build/utils.js:959:15 {
  type: 'Error'
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: Command "yarn run build" exited with 1
```

## Versions

```
node --version
v16.14.2

npm --version
9.1.1

vercel --version
Vercel CLI 28.5.6
```

**Comment by icyJoseph** [comment]
Well with the info given I have to think that the problem is happening when trying to parse `FIREBASE_SERVICE_ACCOUNT_KEY` 

We might ask ourselves... but how, if it is only used server side, or some other argument like that.

I think the `.env-example` file looks weird. If it is supposed to be the `.env` shape, then it should just be something like:

```
NEXT_PUBLIC_REVALIDATE_TOKEN=some-value
NEXT_PUBLIC_FIREBASE_API_KEY=some-other-value
```

And so on... not sure it can be parsed in the current format.

The problem exactly is that, from `about` you import `cities` which lives in `index`, which in turn imports `firebase-admin` utils, which at the top scope of the module does `JSON.parse` so any module that imports `firebase-admin`, directly or indirectly, necessarily runs this top level code.


So my question is, without revealing the actual values, how does your `.env` file look like?

**Comment by jeremyisatrecharm** [comment]
This was happening to me on vercel because I had a `JSON.parse` on app startup of an environment variable that was invalid JSON (I was using single quotes). For example:

```
JSON.parse(process.env.TEST_VAR)
```

```
TEST_VAR = ['one', 'two']
```

JSON only accepts double quotes.
